Still get error "include/liburing/compat.h:25:10: fatal error: 'linux/openat2.h' file not found" when building liburing

Hi Expert, 
Today, based on team's suggestion, download the latest ubuntu image as below
```
ocker images
REPOSITORY                               TAG                                       IMAGE ID      CREATED        SIZE
docker.io/envoyproxy/envoy-build-ubuntu  e2420cc813171c6aad9aa0d62b0544b68bc6504d  36ffed53932d  12 days ago    9.42 GB
```
then fork a main branch and do the building by command "./ci/do_ci.sh  dev.contrib  //contrib/tap_sinks/udp_sink/test:udp_sink_test", got below errors:
```
[11,275 / 17,085] 5 actions, 4 running
ERROR: /source/bazel/foreign_cc/BUILD:28:15: Foreign Cc - Configure: Building liburing failed: (Exit 2): bash failed: error executing CcConfigureMakeRule command (from target //bazel/foreign_cc:liburing) 
  (cd /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy && \
  exec env - \
    BAZEL_CXXOPTS='-stdlib=libc++' \
    BAZEL_LINKLIBS=-l%:libc++.a:-l%:libc++abi.a \
    BAZEL_LINKOPTS=-lm:-pthread \
    CXXFLAGS='-stdlib=libc++' \
    LDFLAGS='-stdlib=libc++ -fuse-ld=lld' \
    PATH=/bin:/usr/bin:/usr/local/bin \
  /bin/bash -c bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/wrapper_build_script.sh)
# Configuration: 09488f0a29b774e35dfcd4117a9e102dd085dcba5a980eddd3de617dfbabcd32
# Execution platform: //bazel/platforms/rbe:linux_x64

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
rules_foreign_cc: Build failed!
rules_foreign_cc: Printing build logs:
_____ BEGIN BUILD LOGS _____

Bazel external C/C++ Rules. Building library 'liburing'

Environment:______________
ENABLE_SHARED=0
BUILD_SCRIPT=bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/build_script.sh
EXT_BUILD_ROOT=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy
BUILD_LOG=bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/Configure.log
PWD=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy
CXXFLAGS=-stdlib=libc++
LDFLAGS=-stdlib=libc++ -fuse-ld=lld
BUILD_WRAPPER_SCRIPT=bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/wrapper_build_script.sh
BAZEL_LINKOPTS=-lm:-pthread
TMPDIR=/tmp
EXT_BUILD_DEPS=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.ext_build_deps
BAZEL_LINKLIBS=-l%:libc++.a:-l%:libc++abi.a
AR=external/llvm_toolchain/bin/llvm-ar
BUILD_TMPDIR=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir
SHLVL=2
ZERO_AR_DATE=1
BAZEL_CXXOPTS=-stdlib=libc++
INSTALLDIR=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing
MAKE=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-opt-exec-ST-d751034c069f/bin/external/rules_foreign_cc/toolchains/private/make/bin/make
PATH=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy:/bin:/usr/bin:/usr/local/bin
RANLIB=external/llvm_toolchain/bin/llvm-ar -s
PKG_CONFIG=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-opt-exec-ST-d751034c069f/bin/external/rules_foreign_cc/toolchains/private/pkgconfig/bin/pkg-config
_=/bin/env
__________________________
Can not copy /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/autoconf
Can not copy /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/automake
Can not copy /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/m4
+ mkdir -p /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing
+ ASFLAGS='-U_FORTIFY_SOURCE --target=x86_64-unknown-linux-gnu -U_FORTIFY_SOURCE -fstack-protector -fno-omit-frame-pointer -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -DABSL_MIN_LOG_LEVEL=4 -Wno-deprecated-declarations -fdebug-types-section -fPIC -fexceptions'
+ CFLAGS='-U_FORTIFY_SOURCE --target=x86_64-unknown-linux-gnu -U_FORTIFY_SOURCE -fstack-protector -fno-omit-frame-pointer -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign --sysroot=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/sysroot_linux_amd64/ -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -DABSL_MIN_LOG_LEVEL=4 -Wno-deprecated-declarations -fdebug-types-section -fPIC -fexceptions'
+ CXXFLAGS='-U_FORTIFY_SOURCE --target=x86_64-unknown-linux-gnu -U_FORTIFY_SOURCE -fstack-protector -fno-omit-frame-pointer -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -std=c++20 -stdlib=libc++ -Xclang -fno-cxx-modules -Wno-module-import-in-extern-c --sysroot=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/sysroot_linux_amd64/ -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -DABSL_MIN_LOG_LEVEL=4 -Wno-deprecated-declarations -fdebug-types-section -fPIC -fsized-deallocation'
+ LDFLAGS='-Wl,-S --target=x86_64-unknown-linux-gnu -no-canonical-prefixes -fuse-ld=lld -lm -Wl,--build-id=md5 -Wl,--hash-style=gnu -Wl,-z,relro,-z,now -l:libc++.a -l:libc++abi.a --sysroot=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/sysroot_linux_amd64/'
+ AR=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/llvm_toolchain/bin/llvm-ar
+ CC=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/llvm_toolchain/bin/cc_wrapper.sh
+ CXX=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/llvm_toolchain/bin/cc_wrapper.sh
+ LD=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/llvm_toolchain/bin/ld.lld
+ RANLIB=:
+ CPPFLAGS=
+ /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/configure --prefix=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing
prefix                             /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing
includedir                         /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/include
libdir                             /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/lib
libdevdir                          /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/lib
relativelibdir                     
mandir                             /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/man
datadir                            /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/share
libgcc_link_flag                   /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/sysroot_linux_amd64/lib/gcc/x86_64-linux-gnu/13/libgcc.a
stringop_overflow                  no
array_bounds                       no
__kernel_rwf_t                     yes
__kernel_timespec                  yes
open_how                           yes
statx                              yes
glibc_statx                        yes
C++                                no
has_ucontext                       yes
has_memfd_create                   yes
NVMe uring command support         no
futex waitv support                no
io_uring discard command support   no
has_idtype_t                       yes
nolibc                             yes
has_fanotify                       yes
ublk_header                        no
use sanitizer                      no
CC                                 /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/llvm_toolchain/bin/cc_wrapper.sh
CXX                                /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/external/llvm_toolchain/bin/cc_wrapper.sh
+ /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-opt-exec-ST-d751034c069f/bin/external/rules_foreign_cc/toolchains/private/make/bin/make install
sed -e "s%@prefix@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing%g" \
    -e "s%@libdir@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/lib%g" \
    -e "s%@includedir@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/include%g" \
    -e "s%@NAME@%liburing%g" \
    -e "s%@VERSION@%2.12%g" \
    liburing.pc.in >liburing.pc
sed -e "s%@prefix@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing%g" \
    -e "s%@libdir@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/lib%g" \
    -e "s%@includedir@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/include%g" \
    -e "s%@NAME@%liburing%g" \
    -e "s%@VERSION@%2.12%g" \
    liburing-ffi.pc.in >liburing-ffi.pc
make[1]: Entering directory '/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/src'
     CC setup.ol
In file included from setup.c:5:
In file included from ./syscall.h:13:
In file included from include/liburing.h:17:
include/liburing/compat.h:25:10: fatal error: 'linux/openat2.h' file not found
   25 | #include <linux/openat2.h>
      |          ^~~~~~~~~~~~~~~~~
1 error generated.
make[1]: *** [Makefile:91: setup.ol] Error 1
make[1]: Leaving directory '/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/src'
make: *** [Makefile:49: install] Error 2
_____ END BUILD LOGS _____
rules_foreign_cc: Build wrapper script location: bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/wrapper_build_script.sh
rules_foreign_cc: Build script location: bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/build_script.sh
rules_foreign_cc: Build log location: bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/Configure.log
rules_foreign_cc: Keeping these below directories for debug, but note that the directories inside a sandbox
rules_foreign_cc: are still cleaned unless you specify the '--sandbox_debug' Bazel command line flag.
rules_foreign_cc: Build Dir: /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir
rules_foreign_cc: Deps Dir: /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/4119/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.ext_build_deps

[11,275 / 17,085] 5 actions running
Target //contrib/exe:envoy-static failed to build
INFO: Elapsed time: 4858.307s, Critical Path: 2088.29s
INFO: 11281 processes: 7154 internal, 4125 linux-sandbox, 1 local, 1 worker.
ERROR: Build did NOT complete successfully
```
thanks
Cliff

Hi @phlax ,
Still has the same error.
what have done:
1) Re base from main branch and include your changes as below
```
   envoy_toolshed = dict(
        project_name = "envoy_toolshed",
        project_desc = "Tooling, libraries, runners and checkers for Envoy proxy's CI",
        project_url = "https://github.com/envoyproxy/toolshed",
        version = "0.3.10",
        sha256 = "4e3ca405ab7990697f75fe75e17f42b3935389afd6fa407eb79700e73de98a11",
        strip_prefix = "toolshed-bazel-v{version}/bazel",
        urls = ["https://github.com/envoyproxy/toolshed/archive/bazel-v{version}.tar.gz"],
        use_category = ["build", "controlplane", "dataplane_core"],
        implied_untracked_deps = [
            "sysroot_linux_amd64",
            "sysroot_linux_arm64",
            "tsan_libs",
            "msan_libs",
        ],
        release_date = "2025-12-11",
```
2) Clearn my build env as below
   bazel clean --expunge
   rm -rf /build/.cache
3) do the build as before
./ci/do_ci.sh dev.contrib //contrib/tap_sinks/udp_sink/test:udp_sink_test

See below for output:
```
ERROR: /source/bazel/foreign_cc/BUILD:28:15: Foreign Cc - Configure: Building liburing failed: (Exit 2): bash failed: error executing CcConfigureMakeRule command (from target //bazel/foreign_cc:liburing) 
  (cd /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy && \
  exec env - \
    BAZEL_CXXOPTS='-stdlib=libc++' \
    BAZEL_LINKLIBS=-l%:libc++.a:-l%:libc++abi.a \
    BAZEL_LINKOPTS=-lm:-pthread \
    CXXFLAGS='-stdlib=libc++' \
    LDFLAGS='-stdlib=libc++ -fuse-ld=lld' \
    PATH=/bin:/usr/bin:/usr/local/bin \
  /bin/bash -c bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/wrapper_build_script.sh)
# Configuration: 09488f0a29b774e35dfcd4117a9e102dd085dcba5a980eddd3de617dfbabcd32
# Execution platform: //bazel/platforms/rbe:linux_x64

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
rules_foreign_cc: Build failed!
rules_foreign_cc: Printing build logs:
_____ BEGIN BUILD LOGS _____

Bazel external C/C++ Rules. Building library 'liburing'

Environment:______________
ENABLE_SHARED=0
BUILD_SCRIPT=bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/build_script.sh
EXT_BUILD_ROOT=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy
BUILD_LOG=bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/Configure.log
PWD=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy
CXXFLAGS=-stdlib=libc++
LDFLAGS=-stdlib=libc++ -fuse-ld=lld
BUILD_WRAPPER_SCRIPT=bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/wrapper_build_script.sh
BAZEL_LINKOPTS=-lm:-pthread
TMPDIR=/tmp
EXT_BUILD_DEPS=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.ext_build_deps
BAZEL_LINKLIBS=-l%:libc++.a:-l%:libc++abi.a
AR=external/llvm_toolchain/bin/llvm-ar
BUILD_TMPDIR=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir
SHLVL=2
ZERO_AR_DATE=1
BAZEL_CXXOPTS=-stdlib=libc++
INSTALLDIR=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing
MAKE=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-opt-exec-ST-d751034c069f/bin/external/rules_foreign_cc/toolchains/private/make/bin/make
PATH=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy:/bin:/usr/bin:/usr/local/bin
RANLIB=external/llvm_toolchain/bin/llvm-ar -s
PKG_CONFIG=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-opt-exec-ST-d751034c069f/bin/external/rules_foreign_cc/toolchains/private/pkgconfig/bin/pkg-config
_=/bin/env
__________________________
Can not copy /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/autoconf
Can not copy /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/automake
Can not copy /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/m4
+ mkdir -p /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing
+ ASFLAGS='-U_FORTIFY_SOURCE --target=x86_64-unknown-linux-gnu -U_FORTIFY_SOURCE -fstack-protector -fno-omit-frame-pointer -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -DABSL_MIN_LOG_LEVEL=4 -Wno-deprecated-declarations -fdebug-types-section -fPIC -fexceptions'
+ CFLAGS='-U_FORTIFY_SOURCE --target=x86_64-unknown-linux-gnu -U_FORTIFY_SOURCE -fstack-protector -fno-omit-frame-pointer -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign --sysroot=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/sysroot_linux_amd64/ -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -DABSL_MIN_LOG_LEVEL=4 -Wno-deprecated-declarations -fdebug-types-section -fPIC -fexceptions'
+ CXXFLAGS='-U_FORTIFY_SOURCE --target=x86_64-unknown-linux-gnu -U_FORTIFY_SOURCE -fstack-protector -fno-omit-frame-pointer -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -std=c++20 -stdlib=libc++ -Xclang -fno-cxx-modules -Wno-module-import-in-extern-c --sysroot=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/sysroot_linux_amd64/ -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -DABSL_MIN_LOG_LEVEL=4 -Wno-deprecated-declarations -fdebug-types-section -fPIC -fsized-deallocation'
+ LDFLAGS='-Wl,-S --target=x86_64-unknown-linux-gnu -no-canonical-prefixes -fuse-ld=lld -lm -Wl,--build-id=md5 -Wl,--hash-style=gnu -Wl,-z,relro,-z,now -l:libc++.a -l:libc++abi.a --sysroot=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/sysroot_linux_amd64/'
+ AR=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/llvm_toolchain/bin/llvm-ar
+ CC=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/llvm_toolchain/bin/cc_wrapper.sh
+ CXX=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/llvm_toolchain/bin/cc_wrapper.sh
+ LD=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/llvm_toolchain/bin/ld.lld
+ RANLIB=:
+ CPPFLAGS=
+ /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/configure --prefix=/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing
prefix                             /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing
includedir                         /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/include
libdir                             /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/lib
libdevdir                          /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/lib
relativelibdir                     
mandir                             /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/man
datadir                            /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/share
libgcc_link_flag                   /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/sysroot_linux_amd64/lib/gcc/x86_64-linux-gnu/13/libgcc.a
stringop_overflow                  no
array_bounds                       no
__kernel_rwf_t                     yes
__kernel_timespec                  yes
open_how                           yes
statx                              yes
glibc_statx                        yes
C++                                no
has_ucontext                       yes
has_memfd_create                   yes
NVMe uring command support         yes
futex waitv support                yes
io_uring discard command support   no
has_idtype_t                       yes
nolibc                             yes
has_fanotify                       yes
ublk_header                        no
use sanitizer                      no
CC                                 /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/llvm_toolchain/bin/cc_wrapper.sh
CXX                                /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/external/llvm_toolchain/bin/cc_wrapper.sh
+ /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-opt-exec-ST-d751034c069f/bin/external/rules_foreign_cc/toolchains/private/make/bin/make install
sed -e "s%@prefix@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing%g" \
    -e "s%@libdir@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/lib%g" \
    -e "s%@includedir@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/include%g" \
    -e "s%@NAME@%liburing%g" \
    -e "s%@VERSION@%2.12%g" \
    liburing.pc.in >liburing.pc
sed -e "s%@prefix@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing%g" \
    -e "s%@libdir@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/lib%g" \
    -e "s%@includedir@%/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/liburing/include%g" \
    -e "s%@NAME@%liburing%g" \
    -e "s%@VERSION@%2.12%g" \
    liburing-ffi.pc.in >liburing-ffi.pc
make[1]: Entering directory '/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/src'
     CC setup.ol
In file included from setup.c:5:
In file included from ./syscall.h:13:
In file included from include/liburing.h:17:
include/liburing/compat.h:25:10: fatal error: 'linux/openat2.h' file not found
   25 | #include <linux/openat2.h>
      |          ^~~~~~~~~~~~~~~~~
1 error generated.
make[1]: *** [Makefile:91: setup.ol] Error 1
make[1]: Leaving directory '/build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir/src'
make: *** [Makefile:49: install] Error 2
_____ END BUILD LOGS _____
rules_foreign_cc: Build wrapper script location: bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/wrapper_build_script.sh
rules_foreign_cc: Build script location: bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/build_script.sh
rules_foreign_cc: Build log location: bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing_foreign_cc/Configure.log
rules_foreign_cc: Keeping these below directories for debug, but note that the directories inside a sandbox
rules_foreign_cc: are still cleaned unless you specify the '--sandbox_debug' Bazel command line flag.
rules_foreign_cc: Build Dir: /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.build_tmpdir
rules_foreign_cc: Deps Dir: /build/.cache/envoy-bazel/bazel_root/base/sandbox/linux-sandbox/7262/execroot/envoy/bazel-out/k8-fastbuild/bin/bazel/foreign_cc/liburing.ext_build_deps

Target //contrib/exe:envoy-static failed to build
INFO: Elapsed time: 9129.753s, Critical Path: 94.00s
INFO: 4262 processes: 8 internal, 4254 linux-sandbox.
ERROR: Build did NOT complete successfully
```

thanks
Cliff

--- a/bazel/foreign_cc/liburing.patch
+++ b/bazel/foreign_cc/liburing.patch
@@ -1,7 +1,44 @@
+diff --git a/configure b/configure
+index 3c95214..1bdf301 100755
+--- a/configure
++++ b/configure
+@@ -118,8 +118,12 @@ print_config() {
+ }
+ 
+ # Default CFLAGS
+-CFLAGS="-D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -include config-host.h"
+-BUILD_CFLAGS=""
++# Preserve original CFLAGS from environment (with sysroot etc)
++ENV_CFLAGS="${CFLAGS}"
++# Add required flags for configure tests
++CFLAGS="${CFLAGS:+$CFLAGS }-D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -include config-host.h"
++# BUILD_CFLAGS will be used by Makefile
++BUILD_CFLAGS="${ENV_CFLAGS}"
+ 
+ # Print configure header at the top of $config_host_h
+ echo "/*" > $config_host_h
+@@ -580,6 +584,8 @@ fi
+ echo "CC=$cc" >> $config_host_mak
+ print_config "CC" "$cc"
+ echo "CXX=$cxx" >> $config_host_mak
++# Write BUILD_CFLAGS so Makefile can use original CFLAGS with sysroot
++echo "BUILD_CFLAGS=$BUILD_CFLAGS" >> $config_host_mak
+ print_config "CXX" "$cxx"
+ 
+ # generate io_uring_version.h
 diff --git a/src/Makefile b/src/Makefile
-index 8f89701a..65439c2f 100644
+index 8f89701..194b6a4 100644
 --- a/src/Makefile
 +++ b/src/Makefile
+@@ -12,7 +12,7 @@ override CPPFLAGS += -D_GNU_SOURCE \
+ 	-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
+ CFLAGS ?= -O3 -Wall -Wextra -fno-stack-protector
+ override CFLAGS += -Wno-unused-parameter \
+-	-DLIBURING_INTERNAL \
++	-DLIBURING_INTERNAL $(BUILD_CFLAGS) \
+ 	$(LIBURING_CFLAGS)
+ SO_CFLAGS=-fPIC $(CFLAGS)
+ L_CFLAGS=$(CFLAGS)
 @@ -44,6 +44,22 @@ else
  endif
  LINK_FLAGS+=$(LDFLAGS)

